.github/workflows/check-automated-doc.yml

```diff
@@ -46,10 +46,10 @@ jobs:
 
       - name: Verify golang generated documentation is up-to-date
         run: |
-          make generate-doc
+          make doc
```

Comment:
As per discussion [here](https://fleetdm.slack.com/archives/C019WG4GH0A/p1734964799330129), this PR changes the Makefile targets:

* generate-doc -> doc
* generate-mock -> mock
* dump-test-schema -> test-schema

The old targets are still there as aliases.